src/Installer/Microsoft.Dotnet.Installation/Internal/ReleaseManifest.cs

```diff
@@ -894,80 +466,42 @@ private static ProductCollection DeserializeProductCollection(string json)
     {
-        var releases = product.GetReleasesAsync().GetAwaiter().GetResult();
-
-        // Get all releases
-        var allReleases = releases.ToList();
+        var releases = product.GetReleasesAsync().GetAwaiter().GetResult().ToList();
 
-        // First try to find the exact version in the original release list
-        var exactReleaseMatch = allReleases.FirstOrDefault(r => r.Version.Equals(resolvedVersion));
-        if (exactReleaseMatch != null)
+        foreach (var release in releases)
         {
-            return exactReleaseMatch;
-        }
-
-        // Now check through the releases to find matching components
-        foreach (var release in allReleases)
-        {
-            bool foundMatch = false;
-
-            // Check the appropriate collection based on the mode
             if (component == InstallComponent.SDK)
             {
                 foreach (var sdk in release.Sdks)
                 {
-                    // Check for exact match
                     if (sdk.Version.Equals(resolvedVersion))
                     {
-                        foundMatch = true;
-                        break;
+                        return sdk;
                     }
-
-                    //  Not sure what the point of the below logic was
-                    //// Check for match on major, minor, patch
-                    //if (sdk.Version.Major == targetReleaseVersion.Major &&
-                    //    sdk.Version.Minor == targetReleaseVersion.Minor &&
-                    //    sdk.Version.Patch == targetReleaseVersion.Patch)
-                    //{
-                    //    foundMatch = true;
-                    //    break;
-                    //}
                 }
             }
-            else // Runtime mode
+            else
             {
-                // Get the appropriate runtime components based on the file patterns
-                var filteredRuntimes = release.Runtimes;
-
-                // Use the type information from the file names to filter runtime components
-                // This will prioritize matching the exact runtime type the user is looking for
-
-                foreach (var runtime in filteredRuntimes)
+                var runtimesQuery = component switch
+                {
+                    InstallComponent.ASPNETCore => release.Runtimes
+                        .Where(r => r.Name.Contains("ASP", StringComparison.OrdinalIgnoreCase)),
+                    InstallComponent.WindowsDesktop => release.Runtimes
+                        .Where(r => r.Name.Contains("Desktop", StringComparison.OrdinalIgnoreCase)),
+                    _ => release.Runtimes
+                        .Where(r => r.Name.Contains(".NET Runtime", StringComparison.OrdinalIgnoreCase) ||
+                               r.Name.Contains(".NET Core Runtime", StringComparison.OrdinalIgnoreCase)),
```

Comment:
Is there a better way to match the right runtime rather than string Contains?

Reply:
I didn't see any other way in the data structure to do so. I agree this approach isn't optimal. We may revisit it when we implement runtime support.